mlir/test/lit.cfg.py

```diff
@@ -247,7 +262,9 @@ def find_real_python_interpreter():
 # TODO: detect Windows situation (or mark these tests as unsupported on these platforms).
 if "asan" in config.available_features:
     if "Linux" in config.host_os:
-        python_executable = f"LD_PRELOAD=$({config.host_cxx} -print-file-name=libclang_rt.asan-{config.host_arch}.so) {config.python_executable}"
+        python_executable = (
+            f"env LD_PRELOAD={get_asan_rtlib()} {config.python_executable}"
```

Comment:
wouldn't it be better to just do `config.environment["LD_PRELOAD"] = get_asan_rtlib()` or does that not work for some reason?

Reply:
The LD_PRELOAD breaks all binary that LLVM has build with asan (like mlir-opt) because those already statically link the asan runtime library. We must only use LD_PRELOAD for `python`.

Reply:
From what I understand, the ASAN library needs to run its constructors before anything else, and that only works by 
- statically linking into the main application (but doesn't work for `python` executable)
- LD_PRELOAD

Reply:
Ok that also makes sense - thanks for the explanation!